Safiyya/circlemapping
# [bug/490] Consolidate name creation from first & last names

### Issue
Related to #490 

### Description
When a user has no surname, their name will appear in some places as "Firstname undefined". This is because when we read data from Auth0 and update names overwriting the data, our logic for creating names is different from when we deserialise users. This is now synchronised to make sure it works the same way.

### Testing
1. Delete your family name in Auth0 metadata
2. Check that when you refresh your profile page in maptio, you can't see your surname
3. On the master branch (ideally) or in production, check that you can see your name in maps as "Firstname undefined"
4. Go to this branch and check that all you see now is "Firstname" (the data will also be corrected in the database on first save)
